engine/src/main/java/org/pentaho/di/trans/steps/userdefinedjavaclass/UserDefinedJavaClassMeta.java

```diff
@@ -139,9 +136,37 @@ public UserDefinedJavaClassMeta() {
       "org.pentaho.di.trans.steps.userdefinedjavaclass.*", "org.pentaho.di.trans.step.*",
       "org.pentaho.di.core.row.*", "org.pentaho.di.core.*", "org.pentaho.di.core.exception.*" } );
 
-    cbe.cook( new Scanner( null, sr ) );
```

Comment:
If the issue is to protect the compilation from throwing unchecked errors, maybe having a try {} catch (Throwable) {} around this line is enough instead of introducing a new Thread